packages/compartment-wrapper/compartment-wrapper.js

```diff
@@ -0,0 +1,70 @@
+export function wrapInescapableCompartment(
+  OldCompartment,
+  inescapableTransforms,
+  inescapableGlobalLexicals,
+) {
+  // This is the new Compartment constructor. We name it `Compartment` so
+  // that it's .name property is correct, but we hold it in 'NewCompartment'
+  // so that lint doesn't think we're shadowing the original.
+  const NewCompartment = function Compartment(
+    endowments,
+    modules,
+    oldOptions = {},
+  ) {
+    const {
+      transforms: oldTransforms = [],
+      globalLexicals: oldGlobalLexicals = {},
+      ...otherOptions
+    } = oldOptions;
+    const newTransforms = [...oldTransforms, ...inescapableTransforms];
+    const newGlobalLexicals = {
+      ...oldGlobalLexicals,
+      ...inescapableGlobalLexicals,
+    };
+    const newOptions = {
+      ...otherOptions,
+      transforms: newTransforms,
+      globalLexicals: newGlobalLexicals,
+    };
+
+    // The real Compartment is defined as a class, so 'new Compartment()'
+    // works but not 'Compartment()'. We can behave the same way. It would be
+    // nice to delegate the 'throw' to the original constructor by knowing
+    // calling it the wrong way, but I don't know how to do that.
+    if (new.target === undefined) {
+      // `newCompartment` was called as a function
+      throw Error('Compartment must be called as a constructor');
+    }
+
+    // It, or a subclass, was called as a constructor
+
+    const c = Reflect.construct(
+      OldCompartment,
+      [endowments, modules, newOptions],
+      new.target,
+    );
+    // The confinement applies to all compartments too. This relies upon the
+    // child's normal Compartment behaving the same way as the parent's,
+    // which will cease to be the case soon (their module tables are
+    // different). TODO: update this when that happens, we need something
+    // like c.globalThis.Compartment = wrap(c.globalThis.Compartment), but
+    // there are details to work out.
+    c.globalThis.Compartment = NewCompartment;
+
+    return c;
+  };
+
+  // ensure `isinstance(c, Compartment)` still holds true
```

Comment:
I see what you did here, `c instanceof Compartment` being the JS spelling of Python’s `isinstance(c, Compartment)`. 😉

Reply:
20 years of python leaves some marks :)